src/Abp.Zero.EntityFramework/EntityHistory/EntityHistoryHelper.cs

```diff
@@ -273,8 +273,8 @@ private ICollection<EntityPropertyChange> GetPropertyChanges(DbEntityEntry entit
                 {
                     propertyChanges.Add(new EntityPropertyChange
                     {
-                        NewValue = isDeleted ? null : propertyEntry.GetNewValue().ToJsonString().TruncateWithPostfix(EntityPropertyChange.MaxValueLength),
-                        OriginalValue = isCreated ? null : propertyEntry.GetOriginalValue().ToJsonString().TruncateWithPostfix(EntityPropertyChange.MaxValueLength),
+                        NewValue = isDeleted && !IsAuditField(propertyEntry.Name) ? null : propertyEntry.GetNewValue().ToJsonString().TruncateWithPostfix(EntityPropertyChange.MaxValueLength),
+                        OriginalValue = isCreated && !IsAuditField(propertyEntry.Name, nameof(IFullAudited.CreatorUserId), nameof(IFullAudited.CreationTime)) ? null : propertyEntry.GetOriginalValue().ToJsonString().TruncateWithPostfix(EntityPropertyChange.MaxValueLength),
```

Comment:
any reason why we need to handle specially `New`/`Original` values for audit field here? 

EntityPropertyChange is created in regardless of audit field or not

Reply:
My idea is to record all the audit fields, and then in the UpdateChangeSet method to determine whether the contents of the audit field have changed, otherwise they will be removed.


CreatorUserId and CreationTime are special values. Their values are unchanged at and after creation.

Reply:
I feel that checking for specific property names in entity history looks like a workaround.

I was trying to wrap my head around this issue. I would propose another solution to address this issue.

1. Add `AuditedAttribute` to the properties in audited entity class (not the interface), e.g FullAudited, CreationAuditedEntity, ...
2. Change post-saving update of entity history from `foreign keys` properties only to `all` properties

Reply:
The current solution also requires that the property name of the user-defined class cannot be the same as the audit name. (When they don't inherit the audit base class.)
If you add the `AuditedAttribute `to the audit base class, you can solve all the problems.

Can you continue to improve it? I think this is faster. 🙂

Reply:
> The current solution also requires that the property name of the user-defined class cannot be the same as the audit name

Yup, that's right. 

> Can you continue to improve it.

I can. I also planning to work on https://github.com/aspnetboilerplate/aspnetboilerplate/pull/4516 😬